pkg/auth/providers/oidc/oidc_client.go

```diff
@@ -18,23 +18,36 @@ func AddCertKeyToContext(ctx context.Context, certificate, key string) (context.
 		}
 		return oidc.ClientContext(ctx, certKeyClient), nil
 	}
+
 	return ctx, nil
 }
 
 func GetClientWithCertKey(httpClient *http.Client, certificate, key string) error {
```

Comment:
It would be more idiomatic to make `GetClientWithCertKey` return `(*http.Client, error)` instead of updating the client.